conans/client/build/cmake.py

```diff
@@ -180,10 +181,11 @@ def command_line(self):
         generator_platform = self.generator_platform
 
         if self.generator_platform and 'Visual Studio' in generator:
-            # FIXME: Conan 2.0 We are adding the platform to the generator instead of using the -A argument
-            #   to keep previous implementation, but any modern CMake will support (and recommend) passing the
-            #   platform in its own argument.
-            compiler_version = self._settings.get_safe("compiler.version")
+            # FIXME: Conan 2.0 We are adding the platform to the generator instead of using
+            #  the -A argument to keep previous implementation, but any modern CMake will support
+            #  (and recommend) passing the platform in its own argument.
+            # Get the version from the generator, as it could have been defined by user argument
+            compiler_version = re.search("Visual Studio ([0-9]*)", generator).group(1)
```

Comment:
I can see the same pattern in the `_build` method, do we need the same change there?

Reply:
Yes, it was extremely unlikely, because it meant using VS 9 2008, but I have changed it too, and added a check in the test.

Reply:
I was referring to line ~320 (inside the function `_build()`):

```
compiler_version = self._settings.get_safe("compiler.version")
```

I think it would be enough to substitute that line with this same regex

Reply:
Not a oneliner, because if the generator is not "Visual Studio", the group() call will raise NoneType. As it is only used for Visual generator, I moved it down to the correct place. I forgot to remove this unused line, I have just fixed it.